python/ray/data/dataset.py

```diff
@@ -2951,19 +2964,31 @@ def iter_torch_batches(
             convert_ndarray_batch_to_torch_tensor_batch,
         )
 
-        for batch in self.iter_batches(
+        def batch_fn(
+            batch: Union[np.ndarray, Dict[str, np.ndarray]]
+        ) -> "TorchTensorBatchType":
+            if collate_fn:
+                batch = collate_fn(batch)
+
+                def tensor_creator_fn(ndarray, dtype, device):
+                    return ndarray.to(dtype=dtype, device=device)
```

Comment:
Is `ndarray` a `torch.Tensor`?

Reply:
Also, why do we need this `tensor_creator_fn`? Like, what's wrong with `convert_ndarray_to_torch_tensor`?

Reply:
no longer needed anymore. removed.